.github/workflows/main.yml

```diff
@@ -0,0 +1,155 @@
+          curl -Lo kamel.tar.gz https://github.com/apache/camel-k/releases/download/1.11.0/camel-k-client-1.11.0-mac-64bit.tar.gz
+          tar -zxvf kamel.tar.gz
+          chmod +x kamel
+          sudo mv kamel /usr/local/bin/
+          kamel install
+          # Clean CLI which were used to start the Kubernetes instance
+          sudo rm /usr/local/bin/kamel
+          sudo rm /usr/local/bin/kubectl
+
+      - name: npm-ci
+        run: npm ci
+
+      - name: npm-vscode:prepublish
+        run: npm run vscode:prepublish
+
+      - name: vsce-package
+        run: vsce package
+
+      - name: test with coverage (ubuntu)
+        id: test_Ubuntu
+        if: matrix.os == 'ubuntu-latest'
+        run: xvfb-run --auto-servernum npm run test:coverage
+
+      - name: test (Macos, Windows)
+        timeout-minutes: 10 # TODO: TMP TIMEOUT, macos test issue
+        if: matrix.os != 'ubuntu-latest'
+        run: npm test
+
+      - name: ui test (Ubuntu)
+        id: uiTest_Ubuntu
+        if: matrix.os == 'ubuntu-latest'
+        run: xvfb-run --auto-servernum npm run ui-test
+
+      - name: ui test (Macos, Windows)
+        id: uiTest_MacOS_Windows
+        if: matrix.os != 'ubuntu-latest'
+        run: npm run ui-test
+
+      - name: Store VS Code Logs (Ubuntu)
+        uses: actions/upload-artifact@v3
+        if: failure() && matrix.os == 'ubuntu-latest'
+        with:
+          name: ${{ matrix.os }}-${{ matrix.version }}-vscode-logs
+          path: ~/.config/Code/logs/*
+
+      - name: Store VS Code Logs (Macos)
+        uses: actions/upload-artifact@v3
+        if: failure() && matrix.os == 'macos-latest'
+        with:
+          name: ${{ matrix.os }}-${{ matrix.version }}-vscode-logs
+          path: ~/Library/Application Support/Code/logs/*
+
+      - name: Store UI test log
+        uses: actions/upload-artifact@v3
+        if: failure() && (steps.uiTest_Ubuntu.outcome == 'failure' || steps.uiTest_MacOS_Windows.outcome == 'failure')
+        with:
+          name: ${{ matrix.os }}-${{ matrix.version }}-ui-test-logs
+          path: test-resources/settings/logs/*
+
+      - name: Store UI Test Screenshots
+        uses: actions/upload-artifact@v3
+        if: failure() && (steps.uiTest_Ubuntu.outcome == 'failure' || steps.uiTest_MacOS_Windows.outcome == 'failure')
+        with:
+          name: ${{ matrix.os }}-${{ matrix.version }}-ui-test-screenshots
+          path: test-resources/screenshots/*.png
+
+      - name: Store Test Coverage
+        uses: actions/upload-artifact@v3
+        if: (steps.test_Ubuntu.outcome == 'success')
+        with:
+          name: ${{ matrix.os }}-${{ matrix.version }}-test-coverage
+          path: coverage/*
+
+      - name: SonarCloud Scan
+        uses: sonarsource/sonarcloud-github-action@master
+        if: matrix.os == 'ubuntu-latest' && github.event_name == 'push'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```

Comment:
I am missing the job which will collect test matrix results and will be used for PR status check

it is OK to not have it in first iteration, just please open the JIRA to implement that right after